dev/TabView/TabViewItem.cpp

```diff
@@ -54,7 +55,11 @@ void TabViewItem::UpdateCloseButton()
         bool canClose = IsCloseable();
         if (auto tabView = SharedHelpers::GetAncestorOfType<winrt::TabView>(winrt::VisualTreeHelper::GetParent(*this)))
         {
-            canClose = canClose && tabView.CanCloseTabs();
+            // IsCloseable defaults to true, but if it hasn't been set then CanCloseTabs should override it.
+            if (!tabView.CanCloseTabs() && ReadLocalValue(IsCloseableProperty()) == winrt::DependencyProperty::UnsetValue())
```

Comment:
I had to stare at this if statement for quite some time to convince myself that it's actually doing the right thing.  Setting canClose to a value and then optionally setting it to false is pretty difficult to parse in terms of what's actually going on here.  Perhaps something like this might be better?

```
auto tabView = SharedHelpers::GetAncestorOfType<winrt::TabView>(winrt::VisualTreeHelper::GetParent(*this));

// We can close a tab in three cases: if IsCloseable has been explicitly set to true;
// if it hasn't been set to false and the tab has no parent to tell it otherwise;
// or if it hasn't been set to false and the tab's parent allows its tabs to close.
bool canClose =
    IsCloseable() &&
    (ReadLocalValue(IsCloseableProperty()) != winrt::DependencyProperty::UnsetValue() ||
        !tabView ||
        tabView.CanCloseTabs());
```

Reply:
As an aside, has this already gone through API review?  As a general rule we're usually encouraged not to rely on "set" vs. "unset" - as an alternative, for example, this could be an enum value like `enum TabViewItemCloseability { Auto, Closeable, NotCloseable }`, where a value of `Auto` means that it inherits from its parent.  This makes it clearly visible to the developer what's going on, and how to tell between "true but that's the default value and can be overridden" and "really super true don't override this".

Reply:
OK, I refactored to be clearer. The API review is next week and you're probably right, this will likely change and I'll make sure that it's brought up.

Reply:
Cool, thanks!